### PR TITLE
[JENKINS-40144] Validate configuration form fields

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-40144](https://issues.jenkins-ci.org/browse/JENKINS-40144)

* Validate all configuration fields
* Fix deprecated API calls
* Fix Incorrect access members on 
* New configuration field private key alias
* Test for the new field
* Change text fields that contain password to password type fields

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java [hunk 1]
```diff
@@ -58,6 +58,7 @@
     private String keystorePath = PAC4J_DEMO_KEYSTORE;
     private String ksPassword = PAC4J_DEMO_PASSWD;
     private String ksPkPassword = PAC4J_DEMO_PASSWD;
+    private String ksPkAlias = "SAML-generated-keyPair";
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java]: Is this a constant? Make it static and final if so.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 2]
```diff
@@ -63,6 +69,22 @@
...
+    public static final String WARN_RECOMMENDED_TO_SET_THE_GROUPS_ATTRIBUTE = "It is recommended to set the groups attribute.";
+    public static final String WARN_RECOMMENDED_TO_SET_THE_USERNAME_ATTRIBUTE = "It is recommended to set the username attribute.";
+    public static final String ERROR_NOT_POSSIBLE_TO_READ_KS_FILE = "It is not possible to read the keystore file.";
+    public static final String ERROR_CERTIFICATES_COULD_NOT_BE_LOADED = "Any of the certificates in the keystore could not be loaded";
+    public static final String ERROR_ALGORITHM_CANNOT_BE_FOUND = "the algorithm used to check the integrity of the keystore cannot be found";
+    public static final String ERROR_NO_PROVIDER_SUPPORTS_A_KS_SPI_IMPL = "No Provider supports a KeyStoreSpi implementation for the specified type.";
+    public static final String ERROR_WRONG_INFO_OR_PASSWORD = "The entry is a PrivateKeyEntry or SecretKeyEntry and the specified protParam does not contain the information needed to recover the key (e.g. wrong password)";
+    public static final String ERROR_INSUFFICIENT_OR_INVALID_INFO = "The specified protParam were insufficient or invalid";
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: i18n?

File: src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java [hunk 3]
```diff
@@ -108,6 +108,27 @@ public void testReadSimpleConfigurationEncryptionData() throws Exception {
...
+            assertEquals(86400, samlSecurityRealm.getMaximumAuthenticationLifetime().longValue());
+            assertEquals("none", samlSecurityRealm.getUsernameCaseConversion());
+            assertEquals("urn:mace:dir:attribute-def:uid", samlSecurityRealm.getUsernameAttributeName());
+            assertEquals(true, samlSecurityRealm.getIdpMetadata().startsWith("<?xml version"));
+            assertEquals("/home/jdk/keystore", samlSecurityRealm.getKeystorePath());
+            assertEquals("changeit", samlSecurityRealm.getKeystorePassword());
+            assertEquals("changeit", samlSecurityRealm.getPrivateKeyPassword());
+            assertEquals("saml-key", samlSecurityRealm.getPrivateKeyAlias());
+        }
+    }
```
Comment (amuniz) [inline src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java]: 👍

File: src/main/java/org/jenkinsci/plugins/saml/OpenSAMLWrapper.java [hunk 4]
```diff
@@ -115,6 +116,7 @@ protected SAML2Client createSAML2Client() {
             config.setKeystorePath(KS.getKeystorePath());
             config.setKeystorePassword(KS.getKsPassword());
             config.setPrivateKeyPassword(KS.getKsPkPassword());
+            config.setKeystoreAlias(KS.getKsPkAlias());
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/OpenSAMLWrapper.java]: Is this related to validations?

File: src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java (hunk 1 above)
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java]: there is a mistake here.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java (hunk 2 above)
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: I do not plan to support i18n, is only to keep messages on constants.

File: src/main/java/org/jenkinsci/plugins/saml/OpenSAMLWrapper.java (hunk 4 above)
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/OpenSAMLWrapper.java]: without the key alias the validation of the keystore is a little odd, even do it is tested without the key alias but do not have the key alias forces you to have an only one certificate keystore
